src/transport_ofi.h

```diff
@@ -554,18 +557,20 @@ void shmem_transport_get_wait(void)
         success = fi_cntr_read(shmem_transport_ofi_get_cntrfd);
         fail = fi_cntr_readerr(shmem_transport_ofi_get_cntrfd);
 
-        if (success < shmem_transport_ofi_pending_get_counter && fail == 0) {
+        if (success < \
+            shmem_internal_atomic_read(&shmem_transport_ofi_pending_get_counter) && \
+            fail == 0) {
             SPINLOCK_BODY();
         }
         else if (fail) {
             struct fi_cq_err_entry e = {0};
             fi_cq_readerr(shmem_transport_ofi_put_nb_cqfd, (void *)&e, 0);
             OFI_CQ_ERROR(shmem_transport_ofi_put_nb_cqfd, &e);
         }
-    } while (success < shmem_transport_ofi_pending_get_counter);
+    } while (success < shmem_internal_atomic_read(&shmem_transport_ofi_pending_get_counter));
 #else
     int ret = fi_cntr_wait(shmem_transport_ofi_get_cntrfd,
-                           shmem_transport_ofi_pending_get_counter, -1);
+                           shmem_internal_atomic_read(&shmem_transport_ofi_pending_get_counter), -1);
```

Comment:
Same comment as above, we need to check if the pending counter increased while we were blocked.